### ISSUE TITLE
ncclient edit_config breaks on IOS-XE 17.3

### ISSUE THREAD

**Issue by dmfigol** [ncclient/ncclient#489]
Hi,
I have the following problem on Cisco CSR1000v running 17.3.1a (min repro)
**UPD**: workaround is at the end of the post
```python
CFG = """
<config>
  <native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
    <hostname>R1</hostname>
  </native>
</config>
"""

CONN_DATA = {
        "host": "192.168.152.101",
        "username": "cisco",
        "password": "cisco,
        "hostkey_verify": False,
}

with manager.connect(**CONN_DATA) as nc_conn:
    nc_response = nc_conn.edit_config(config=CFG, target="running")
```
Output:
```
  File "/home/dmfigol/projects/network-programmability-stream/scrapli-apps/10-csr-nc-ncclient.py", line 53, in nc_get_edit_cfg
    nc_response = nc_conn.edit_config(config=CFG, target="running")
  File "/home/dmfigol/projects/network-programmability-stream/scrapli-apps/.venv/lib/python3.9/site-packages/ncclient/manager.py", line 231, in execute
    return cls(self._session,
  File "/home/dmfigol/projects/network-programmability-stream/scrapli-apps/.venv/lib/python3.9/site-packages/ncclient/operations/edit.py", line 69, in request
    return self._request(node)
  File "/home/dmfigol/projects/network-programmability-stream/scrapli-apps/.venv/lib/python3.9/site-packages/ncclient/operations/rpc.py", line 348, in _request
    raise self._reply.error
ncclient.operations.rpc.RPCError: {'type': 'protocol', 'tag': 'unknown-element', 'severity': 'error', 'info': '<?xml version="1.0" encoding="UTF-8"?><error-info xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0"><bad-element>config</bad-element>\n</error-info>\n', 'path': '\n    /rpc/edit-config\n  ', 'message': None}
```

Using logs I was able to track down sent payload:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:c7da5fdc-17c4-44f4-bb0c-ae4a10de8d43">
<nc:edit-config>
    <nc:target>
        <nc:running/>
    </nc:target>
    <config>
        <native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
            <hostname>R1-1</hostname>
        </native>
    <config>
</nc:edit-config>
</nc:rpc>
```
which fails when applied on netconf session directly. However, if I add namespace prefix to `config` element, then it is fine:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:c7da5fdc-17c4-44f4-bb0c-ae4a10de8d43">
<nc:edit-config>
    <nc:target>
        <nc:running/>
    </nc:target>
    <nc:config>
        <native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
            <hostname>R1-1</hostname>
        </native>
    </nc:config>
</nc:edit-config>
</nc:rpc>
```
reply:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:c7da5fdc-17c4-44f4-bb0c-ae4a10de8d43" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0"><ok/></rpc-reply>]]>]]>
```

I am not 100% sure if the device should be able to handle that per RFC, at the same time I am not sure why `nc:` prefix is not added to `config` element by `ncclient`.

**UPD**: workaround for people stumbling upon this, either add `nc:` prefix to `config` element or add a default namespace, e.g.:
```xml
<config xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
    <hostname>R1</hostname>
  </native>
</config>
```
or
```xml
<nc:config>
  <native xmlns="http://cisco.com/ns/yang/Cisco-IOS-XE-native">
    <hostname>R1</hostname>
  </native>
</nc:config>
```

**Comment by einarnn** [comment]
@dmfigol, the device is not obligated to accept the non-namespaced `config` element. A number of vendor platforms (yes, IOS XE and IOS XR included) have, in the past, been lax in their checking of the namespace of some elements. IOS XE 17.3.1 is an example of where a platform has tightened up parsing and which now insists on having the correct namespace for the config element.

`ncclient` has never automatically added any namespaces, but a valid tweak to `edit_config` may be to add the netconf base 1.0 namespace to the config element if it's not provided by the caller.

**Comment by dmfigol** [comment]
@einarnn well, ncclient does add `nc:` prefix to `rpc`, `edit-config`. AFAIK, without any prefixes, config would be accepted as is.
I fixed it with :
```
        default_ns = "urn:ietf:params:xml:ns:netconf:base:1.0"
        root_element = etree.Element("config", nsmap={None: default_ns}) 
```
but really wish it wasn't necessary

**Comment by einarnn** [comment]
> @einarnn well, ncclient does add nc: prefix to rpc, edit-config. AFAIK, without any prefixes, config would be accepted as is.

This is different -- `ncclient` is creating that content, not _modifying what a developer has passed in_.

The case of the config element is a restricted case, and I think we can contemplate it being added by `ncclient`. I will try and look at that soon if no-one else does.

**Comment by dev-Rhllor** [comment]
+1
I hit hard with this issue. I think the workaround proposed by @dmfigol makes a lot of sense and will increase the user experience of the Average Joe network developer.

**Comment by einarnn** [comment]
Addressed in v0.6.13, via PR #528
